src/vendor/vscode-ocaml-platform/src-bindings/vscode_languageclient/vscode_languageclient.ml

```diff
@@ -190,6 +190,38 @@ module StaticFeature = struct
       [@@js.builder]]
 end
 
+module StreamInfo = struct
+  include Interface.Make ()
+
+  type njs_stream
+
+  let njs_stream_of_string str : njs_stream =
+    Format.ksprintf
+      Js_of_ocaml.Js.Unsafe.eval_string
+      "new joo_global_object.webSocket (`%s`);"
```

Comment:
```suggestion
      "new webSocket (`%s`);"
```
(maybe … currently testing).

Now I get `invalid header: GET / HTTP/1.1` from the server. But I guess that's progress :-)

Reply:
From what I'm reading[^1], there should be no need to use websockets for this as the protocol is not "over HTTP". Here regular sockets should be used; just hoping there's not too many Ojs bindings to write…

[^1]: *cf* https://github.com/microsoft/vscode-languageserver-node/issues/662

Reply:
Or else an "HTTP" mode needs to be implemented on the LSP server side.

Reply:
Pursuing on this, I think there's a way to just bind [net.connect](https://nodejs.org/api/net.html#netconnectport-host-connectlistener) in `src/vendor/vscode-ocaml-platform/src-bindings/node/node.ml[i]`.

Reply:
It's already there actually: `Net.Socket.connect` does the same as `net.connect`.

Reply:
Indeed. Did you try already? (I may try if you have not and/or cannot); I'm just not sure how to adapt `LanguageClient.make_stream`, or even if that's needed.

Reply:
I am updating it at the moment :)

Reply:
I get the same "invalid header" error with the net socket version... But it's one less dependency, so I'll commit this version

Reply:
I think I've just had it running with hardcoded "localhost" and port number, removing the WebSocket instantiation, and one more ugly `Obj.magic` ;-)

Reply:
You mean with a simple `LanguageClient.make`?

Reply:
The idea is to still go via `StreamInfo`, just casting the socket onto that as I guess it also has the expected  `read`/`write` operations. Then instead of `ws://`prefix (it's not websockets anymore), I'd advocate for a match against a simple `host:port` regexp (no `/` in `host`, only digits in `port`); that should be enough I guess..

Reply:
I can't understand why your version works and "documentation" (github projets + chatgpt) sends me in totally different directions. Can you push your version so that I can understand what's going on?

Reply:
Sure! You can check that [here](https://github.com/nberth/superbol-studio-oss/tree/remote-lsp).

Reply:
It works!